### PR TITLE
implemented similarity distance functions

### PR DESCRIPTION
**Pull request by lunasara**
I've created this

### CODE REVIEW

File: distances.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        hist_im_query: Matrix containing the test image histogram data
+
+    Returns: Matrix resizing the test image histogram data
+
+    """
+    nim_dataset, nvals_dataset = np.shape(hist_dataset)
+    nims, nvals_query = np.shape(hist_im_query)
+
+    if nvals_query != nvals_dataset:
+        print("Dimensions don't match")
```
Comment (yaeltudela) [inline distances.py]: should raise an exception

File: distances.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+    hist_im_query = np.tile(hist_im_query, (nim_dataset, 1))
```
Comment (yaeltudela) [inline distances.py]: making `hist_im_query = hist_im_query[..., None] ` adds a dimension to array, making possible to broadcast on dataset_hist

File: distances.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    if mode == 'distance_x2':
+        return distance_x2(hist_dataset, hist_im_query)
+    if mode == 'intersection':
+        return intersection(hist_dataset, hist_im_query)
+    if mode == 'kl_divergence':
+        return kl_divergence(hist_dataset, hist_im_query)
+    if mode == 'js_divergence':
+        return js_divergence(hist_dataset, hist_im_query)
+    if mode == 'hellinger':
+        return hellinger(hist_dataset, hist_im_query)
```
Comment (yaeltudela) [inline distances.py]: Use if ... elif .. else strucutre.
Add NotImplementedError exception on else

File: distances.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
+import numpy as np
+
+
+def values(hist_dataset, hist_im_query):
```
Comment (yaeltudela) [inline distances.py]: rename to broadcast_query_values or remove and replace with `hist_im_query = hist_im_query[..., None]` inside every function

**Review by yaeltudela — CHANGES_REQUESTED** [review]

File: distances.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
+import numpy as np
+    
+
+def values(hist_dataset, hist_im_query_ini):
```
Comment (lunasara) [inline distances.py:4]: I've changed this function completely so it verifies the size of the histograms to be compared

**Pull request merged**